From 4616962f2de22e1b12263cdf4439db3d40257879 Mon Sep 17 00:00:00 2001
From: Jeegar Patel <jeegar.patel@xilinx.com>
Date: Tue, 8 Sep 2020 01:37:32 -0700
Subject: [PATCH 1574/1851] v4l: xilinx: dma: Fix INFINITE_LOOP warning

commit 6457cbe6a8dcb0b1d5cf9b3924ccd84e4ced17c2 from
https://github.com/Xilinx/linux-xlnx.git

This patch fix INFINITE_LOOP type coverity warning.

warning:
Loop bound "*nplanes" has type "unsigned int", which is wider in
size or has a bigger upper bound than the type "u8" of loop counter
"i", the loop condition may always be true.

Signed-off-by: Jeegar Patel <jeegar.patel@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-dma.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/platform/xilinx/xilinx-dma.c b/drivers/media/platform/xilinx/xilinx-dma.c
index 98bd568fd909..4f234113babc 100644
--- a/drivers/media/platform/xilinx/xilinx-dma.c
+++ b/drivers/media/platform/xilinx/xilinx-dma.c
@@ -323,7 +323,7 @@ xvip_dma_queue_setup(struct vb2_queue *vq,
 		     unsigned int sizes[], struct device *alloc_devs[])
 {
 	struct xvip_dma *dma = vb2_get_drv_priv(vq);
-	u8 i;
+	unsigned int i;
 	int sizeimage;
 
 	/* Multi planar case: Make sure the image size is large enough */
-- 
2.31.1

